static/src/javascripts/projects/common/modules/identity/ad-prefs.js

```diff
@@ -80,31 +73,100 @@ class ConsentBox extends Component<
                     <ConsentRadioButton
                         label="Turn on"
                         value="true"
-                        checked={this.state.consentState === true}
+                        checked={this.props.state === true}
                         consent={this.props.consent}
-                        onToggle={() => this.setConsentTo(true)}
+                        onToggle={() => this.props.onUpdate(true)}
                     />
                     <ConsentRadioButton
                         label="Turn off"
                         value="false"
-                        checked={this.state.consentState === false}
+                        checked={this.props.state === false}
                         consent={this.props.consent}
-                        onToggle={() => this.setConsentTo(false)}
+                        onToggle={() => this.props.onUpdate(false)}
                     />
                 </div>
             </fieldset>
         );
     }
 }
 
-class ConsentBoxes extends Component<{}, {}> {
+class ConsentBoxes extends Component<
+    {},
+    { consentsWithState: AdConsentWithState[], changesPending: boolean }
+> {
+    constructor(props: {}) {
+        super(props);
+        this.state = {
+            changesPending: false,
+            consentsWithState: allAdConsents.map((consent: AdConsent) => ({
+                consent,
+                state: getAdConsentState(consent),
+            })),
+        };
+    }
+
+    onUpdate(consentId, state: AdConsentState) {
+        const consentsWithState = { ...this.state }.consentsWithState;
```

Comment:
Is destructuring necessary here? Could we not just write `this.state. consentsWithState`.

Reply:
```js
const { consentsWithState } = this.state;
```
?

Reply:
I don't want to mutate the original state, just to grab a copy of it. I know it seems a bit pointless considering I'm using the copy to define the new state but I believe this is best practice